src/routes/PetsPage/PetsPage.js

```diff
@@ -0,0 +1,40 @@
+import React, { Component } from 'react';
+import PetCard from '../../components/PetCard/PetCard';
+import PetsService from '../../services/petsService';
+
+export default class PetsPage extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { pets: [] };
+  }
+
+  componentDidMount() {
+    PetsService.getPets()
+      .then(pets => this.setState({ pets }))
+      .catch(error => console.log(error));
+  }
+
+  renderPetRow(rowStart) {
+    return (
+      <div className='row'>
+        {this.state.pets.slice(rowStart, rowStart + 3).map(pet => this.renderPet(pet))}
+      </div>
+    );
+  }
+
+  renderPet(pet) {
+    return (
+      <PetCard key={pet.PetID} pet={pet} />
+    );
+  }
+  
+  render() {
+    const { pets } = this.state;
+    const petRows = pets.length / 3 + (pets.length % 3 === 0 ? 0 : 1);
```

Comment:
- When there are no data, the page is just a blank page. It's probably nice to have a message instead of having a blank page.
- It looks like the page works only when the pets.length is multiples of 3. I do think that having three items in a row is good. I think you just need to change the calculation here.